lib/disk/modules/RawBlockIO.rb

```diff
@@ -1,14 +1,10 @@
-
-$:.push("#{File.dirname(__FILE__)}")
-
-require 'io/extra'
-require 'MiqBlockDevOps'
+require_relative 'MiqBlockDevOps'
 
 class RawBlockIO
 
   MIN_SECTORS_TO_CACHE = 64
 
-  def initialize(filename, mode="r")
+  def initialize(filename, mode=File::RDONLY|File::DIRECT)
```

Comment:
No problem, we can use fcntl:

    file = 'temp.txt'
    fh = File.open(file, 'r')
    fh.fcntl(Fcntl::F_SETFL, File::DIRECT)
    p fh.fcntl(Fcntl::F_GETFL) # => 49152
    fh.close